instagram/src/App.js

```diff
@@ -10,17 +10,42 @@ class App extends Component {
   constructor(){
     super();
     this.state={
-      data : dummyData,
+      data : [],
+      bully: false,
     }
   }
+
+  componentDidMount(){
+    this.setState({data:dummyData}, ()=>{console.log(this.state.data)})
+    // setTimeout(()=> {alert('Hello') this.setState({bully:true}, console.log(this.state.bully)) }, 3000)
+  }
+
+  //always conditional, causes rerender
+  // componentDidUpdate(){
+  //   if(this.state.bully){
+  //     console.log('Would Rerender')
+  //   }
+  // }
+
+  componentWillUnmount(){
+    clearTimeout();
```

Comment:
from what I can tell, you were using this to account for a still-pending timeout function from the CDM in case the user refreshes the page before it actually loads--a good case to account for actually, but without the setTimeout it's not necessary to leave this in